python_modules/dagster/dagster/_core/definitions/asset_selection.py

```diff
@@ -415,6 +415,26 @@ def replace(self, **kwargs):
         return func(self, update=kwargs)
 
 
+@whitelist_for_serdes
+class AllAutoMaterializableSelection(AssetSelection, frozen=True):
+    def resolve_inner(self, asset_graph: AssetGraph) -> AbstractSet[AssetKey]:
+        return {
+            target_key
+            for target_key in asset_graph.materializable_asset_keys
+            if asset_graph.get_auto_materialize_policy(target_key) is not None
+        }.union(
+            target_key
+            for target_key in asset_graph.source_asset_keys
+            if asset_graph.get_auto_observe_interval_minutes(target_key) is not None
+        )
+
+    def to_serializable_asset_selection(self, asset_graph: AssetGraph) -> "AssetSelection":
+        return self
+
+    def __str__(self) -> str:
+        return "all auto-materializable assets and auto-observable source assets"
```

Comment:
I think we should capitalize "All" since I just put this string wholesale into the UI